model/riscv_sys_control.sail

```diff
@@ -342,9 +342,9 @@ function dispatchInterrupt(priv : Privilege) -> option((InterruptType, Privilege
   if (~ (haveUsrMode())) | ((~ (haveSupMode())) & (~ (haveNExt()))) then {
     assert(priv == Machine, "invalid current privilege");
```

Comment:
Looking at this again this assert looks suspect. If `haveUsrMode()` is `true` but `haveSupMode()` and `haveNExt()` are false this could be reachable with `priv == User`. I think the second half of the OR in the condition should go away.

Reply:
A comment from @pmundkur would be ideal.

Reply:
I think I've figured it out: the condition is checking whether it is possible for any interrupts to be delegated to a mode other than Machine mode. This makes sense as it then uses `mip` and `mie` directly to compute the pending interrupts rather than calling `getPendingSet` (which considers delegation). This means we need to fix or remove the assertion. I think probably we should remove it as I can't think of much to put there that isn't just a rehash of the preciding if condition.

On the subject of that condition I think it could also be expressed as `~(haveSupMode()) | (haveUsrMode() & ~haveNExt())` which might more clearly express the intent?

Reply:
> On the subject of that condition I think it could also be expressed as ~(haveSupMode()) | (haveUsrMode() & ~haveNExt()) which might more clearly express the intent?

The above is incorrect but a more concise version might be `~(haveSupMode() | haveNExt())`?